tests/unit/s2n_server_certificate_status_test.c

```diff
@@ -158,7 +158,7 @@ int main(int argc, char **argv)
         EXPECT_SUCCESS(s2n_stuffer_growable_alloc(&stuffer, 0));
 
         EXPECT_SUCCESS(s2n_tls13_server_status_request_extension.send(conn, &stuffer));
-        EXPECT_FAILURE_WITH_ERRNO(s2n_tls13_server_status_request_extension.recv(conn, &stuffer), S2N_ERR_CERT_UNTRUSTED);
+        EXPECT_FAILURE(s2n_tls13_server_status_request_extension.recv(conn, &stuffer));
 
```

Comment:
This still really needs to be a "WITH_ERRNO". Don't you expect a specific error?

Reply:
I switched a few of these to just EXPECT_FAILURE because in refactoring the validator functions I used macros like `RESULT_GUARD_POSIX` and `RESULT_ENSURE_REF`, which don't allow you to override the error.

In dd6b143db08d592ed8485c793845495093d4bbbe I changed all the tests back to use EXPECT_FAILURE_WITH_ERRNO, and changed the macros to ones that allow you to set the error.